io.openems.edge.evcs.keba.kecontact/test/io/openems/edge/evcs/keba/kecontact/EvcsKebaKeContactImplTest.java

```diff
@@ -1,27 +1,167 @@
 package io.openems.edge.evcs.keba.kecontact;
 
+import static org.junit.Assert.assertEquals;
+
 import org.junit.Test;
 
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.common.types.ChannelAddress;
+import io.openems.edge.common.channel.IntegerReadChannel;
+import io.openems.edge.common.test.AbstractComponentTest.TestCase;
 import io.openems.edge.common.test.ComponentTest;
+import io.openems.edge.common.test.DummyComponentManager;
+import io.openems.edge.evcs.api.Evcs;
+import io.openems.edge.evcs.api.ManagedEvcs;
+import io.openems.edge.evcs.api.Phases;
+import io.openems.edge.evcs.api.Status;
 import io.openems.edge.evcs.keba.kecontact.core.EvcsKebaKeContactCoreImpl;
 import io.openems.edge.evcs.test.DummyEvcsPower;
 
 public class EvcsKebaKeContactImplTest {
 
 	private static final String COMPONENT_ID = "evcs0";
+	private EvcsKebaKeContactImpl evcs;
+
+	// Channel Addresses EVCS0
+	private static ChannelAddress status = new ChannelAddress("evcs0", Evcs.ChannelId.STATUS.id());
+	private static ChannelAddress chargePower = new ChannelAddress("evcs0", Evcs.ChannelId.CHARGE_POWER.id());
+	private static ChannelAddress phases = new ChannelAddress("evcs0", Evcs.ChannelId.PHASES.id());
+	private static ChannelAddress minimumHardwarePower = new ChannelAddress("evcs0",
+			Evcs.ChannelId.MINIMUM_HARDWARE_POWER.id());
+	private static ChannelAddress maximumHardwarePower = new ChannelAddress("evcs0",
+			Evcs.ChannelId.MAXIMUM_HARDWARE_POWER.id());
+	private static ChannelAddress fixedMaximumHwPower = new ChannelAddress("evcs0",
+			Evcs.ChannelId.FIXED_MAXIMUM_HARDWARE_POWER.id());
+	private static ChannelAddress chargingstationCommunicationFailed = new ChannelAddress("evcs0",
+			Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED.id());
+
+	private static ChannelAddress setChargePowerLimit = new ChannelAddress("evcs0",
+			ManagedEvcs.ChannelId.SET_CHARGE_POWER_LIMIT.id());
 
 	@Test
-	public void test() throws Exception {
-		new ComponentTest(new EvcsKebaKeContactImpl()) //
+	public void testAdjustPowerLimitsOnPhaseSwitching() throws OpenemsException, Exception {
+
+		this.evcs = new EvcsKebaKeContactImpl();
```

Comment:
fyi: Deine auskommentierten Tests werfen alle nullpointer exceptions, weil du this.evcs nicht neu setzt.
du kannst wenn du mehrere Tests hast mit der Annotation @before arbeiten.
z.B. 
`@Before
	public void setUp() throws OpenemsException, Exception {
		this.evcs = new EvcsKebaKeContactImpl();
		this.test = new ComponentTest(this.evcs) //
				.addReference("evcsPower", new DummyEvcsPower()) //
				.addReference("componentManager", new DummyComponentManager()) //
				.addReference("kebaKeContactCore", new EvcsKebaKeContactCoreImpl()) //
				.activate(MyConfig.create() //
						.setId(COMPONENT_ID) //
						.setDebugMode(false) //
						.setIp("172.0.0.1") //
						.setMinHwCurrent(6000) //
						.setUseDisplay(false) //
						.setPhaseSwitchActive(true) //
						.build());
	}`
	dann wird vor jedem deiner Tests this.evcs gesetzt und du läufst in keine nullpointer

Reply:
Hallo Johann,

kannst du das eventuell hier in dem PR pushen ?

Das wäre der absolute Hammer ! :)

Grüße !

Reply:
Hallo Johann,

gibt es hier schon was neues ?

Grüße !

Reply:
Hallo sn0w3y,

ich war bis heute im Urlaub, dementsprechend noch nichts neues.
was meinst du mit hier pushen? Spezifisch das mit "@Before"?

Reply:
Hallo @johannk24 ,

nene, ich dachte du hättest schon eine Version :)